auto-dark.el

```diff
@@ -26,6 +26,8 @@
 
 
 ;;; Code:
+(require 'dbus)
```

Comment:
Hello there! Thanks for your prompt response!

Indeed this solves the problem, but I think a top general require for something that is not "core" for all OSs might be a bit off (stylistic speaking).

Let me elaborate a little :)

I was thinking on "isolate" this require to the `gnu/linux` portion of the code. Specifically on the entry function here: https://github.com/LionyxML/auto-dark-emacs/blob/master/auto-dark.el#L78

Maybe:
```elisp
(defun auto-dark--is-dark-mode-dbus ()
  "Use Emacs built-in D-Bus function to determine if dark theme is enabled."
  (when (require 'dbus nil t)
    (eq 1 (caar (dbus-call-method
		 :session
		 "org.freedesktop.portal.Desktop"
		 "/org/freedesktop/portal/desktop"
                 "org.freedesktop.portal.Settings" "Read"
                 "org.freedesktop.appearance" "color-scheme")))))
``` 
This would:
- Require the dbus package only on a condition it might be present (gnu/linux)
- Stop the code execution if the package is not present and free us of dealing with some exception specific for a user case that have a window manager / system configured in a non conventional "d-bus way".

How do you feel about this approach?

Reply:
My primary concern with this approach is that there are four locations in this library where a `dbus-*` function is called.  This one is&mdash;as the code is currently written&mdash;likely to be the first invoked, but if the code were to be rearranged such that a different `dbus-*` invocation were to be called first, it must also be guarded with `(when (require 'dbus nil t)` lest we have this same problem again.  Thus, it seems that the future-proof approach would be to guard all invocations of `dbus-*` function with `(when (require 'dbus nil t)`, which also seems a bit heavy-handed.  As I was typing and thinking about this, I came up with what I suspect to be a slightly more elegant approach, which I'll try in a different PR (which may supersede this one, if you're in favor of the change).

---

As a bit of tangent, the `dbus` library exists in all versions of Emacs I've tried, including macOS and Windows versions.  That said, there is a compiler flag to not include it: `--without-dbus`, thus it's possible that _any_ version of Emacs&mdash;for macOS, Windows, or Linux&mdash;is compiled with this flag, excluding the library.